appengine/js/js-blocks.js

```diff
@@ -272,8 +346,159 @@ Blockly.Blocks['math_change'].init = function() {
+    "helpUrl": Blockly.Msg.MATH_RANDOM_INT_HELPURL
+  });
+};
+
 Blockly.Msg.MATH_RANDOM_FLOAT_TITLE_RANDOM = 'Math.random  (  )';
 
+Blockly.Msg.LISTS_CREATE_EMPTY_TITLE = '[ ]';
+Blockly.Msg.LISTS_CREATE_WITH_INPUT_WITH = '[';
+
+/**
+ * Modify create list with elements block to have the correct number of inputs.
+ * @private
+ * @this Blockly.Block
+ */
+Blockly.Blocks['lists_create_with'].updateShape_ = function() {
+  if(this.getInput('TAIL')) {
+    this.removeInput('TAIL');
+  }
+  if (this.itemCount_ && this.getInput('EMPTY')) {
+    this.removeInput('EMPTY');
+  } else if (!this.itemCount_ && !this.getInput('EMPTY')) {
+    this.appendDummyInput('EMPTY')
+        .appendField(Blockly.Msg.LISTS_CREATE_EMPTY_TITLE);
+  }
+  // Add new inputs.
+  for (var i = 0; i < this.itemCount_; i++) {
+    if (!this.getInput('ADD' + i)) {
+      var input = this.appendValueInput('ADD' + i);
+      if (i == 0) {
+        input.appendField(Blockly.Msg.LISTS_CREATE_WITH_INPUT_WITH);
+      } else {
+        input.appendField(',');
+      }
+    }
+  }
+  // Remove deleted inputs.
+  while (this.getInput('ADD' + i)) {
+    this.removeInput('ADD' + i);
+    i++;
+  }
+  if(this.itemCount_) {
+    this.appendDummyInput('TAIL')
+        .appendField(']');
+  }
+};
+
+Blockly.Blocks['lists_getIndex'] = {
+  /**
+   * Block for getting element at index.
+   * @this Blockly.Block
+   */
+  init: function () {
+    this.jsonInit({
+      "message0": "%1[%2]",
+      "args0": [
+        {
+          "type": "input_value",
+          "name": "VALUE",
+          "check": "Array"
+        },
+        {
+          "type": "input_value",
+          "name": "AT",
+          "check": "Number"
+        }
+      ],
+      "inputsInline": true,
+      "output": null,
+      "colour": Blockly.Blocks.lists.HUE,
+      "tooltip": Blockly.Msg.LISTS_GET_INDEX_TOOLTIP_GET_FROM +
+          Blockly.Msg.LISTS_INDEX_FROM_START_TOOLTIP.replace('%1', '#0'),
+      "helpUrl": Blockly.Msg.LISTS_GET_INDEX_HELPURL
+    });
+  }
+};
+
+Blockly.Blocks['lists_setIndex'] = {
+  /**
+   * Block for setting element at index.
+   * @this Blockly.Block
+   */
+  init: function () {
+    this.jsonInit({
+      "message0": "%1[%2] = %3;",
+      "args0": [
+        {
+          "type": "input_value",
+          "name": "LIST",
+          "check": "Array"
+        },
+        {
+          "type": "input_value",
+          "name": "AT",
+          "check": "Number"
+        },
+        {
+          "type": "input_value",
+          "name": "TO"
+        }
+      ],
+      "inputsInline": true,
+      "previousStatement": null,
+      "nextStatement": null,
+      "colour": Blockly.Blocks.lists.HUE,
+      "tooltip": Blockly.Msg.LISTS_SET_INDEX_TOOLTIP_SET_FROM +
+          Blockly.Msg.LISTS_INDEX_FROM_START_TOOLTIP.replace('%1', '#0'),
+      "helpUrl": Blockly.Msg.LISTS_SET_INDEX_HELPURL
+    });
+  }
+};
+
+Blockly.Msg.LISTS_LENGTH_TITLE = '%1 . length';
```

Comment:
The ? is supposed to be replaced with the variable chosen for the first argument. It is not implemented. I've added a TODO.